### ISSUE TITLE
--lineaftermarks not adding spacing above // MARK

### ISSUE THREAD

**Issue by tciuro** [nicklockwood/SwiftFormat#1758]
According to the [documentation](https://github.com/nicklockwood/SwiftFormat/blob/main/Rules.md#blankLinesAroundMark), `--lineaftermarks` does the following:

> Insert a blank line after multiline switch cases (excluding the last case, which is followed by a closing brace).

I have enabled the `--lineaftermarks` rule like so:

```
--lineaftermarks true
```

The code looks like this:

```swift
        .navigationBarItems(trailing: SomeTrailingItem(contentViewModel: myViewModel, navigationRouter: myNavigationRouter))
        // MARK: Analytics

```

when I run swiftformat, I expected the code to be modified like so:

```swift
        .navigationBarItems(trailing: SomeTrailingItem(contentViewModel: myViewModel, navigationRouter: myNavigationRouter))

        // MARK: Analytics

```

However, the code is left untouched. Not only that, when I add spacing myself, it reverts the change and removes the added line.

[Update]

Even if I set `--lineaftermarks false`, it stubbornly removes the line.

Xcode: 15.4
swiftformat version: 0.54.1

**Comment by nicklockwood** [comment]
What does the line after the MARK comment look like?

**Comment by tciuro** [comment]
Hi @nicklockwood !

```swift
        .navigationBarItems(trailing: SomeTrailingItem(contentViewModel: myViewModel, navigationRouter: myNavigationRouter))
        // MARK: Analytics

        .onChange(of: yesSelected) { newValue in
```

**Comment by tciuro** [comment]
I also tried to play with the setting: `--enable blankLinesAroundMark`, but no combination seems to work.

**Comment by nicklockwood** [comment]
OK so this is probably due to the [blanklinesbetweenchainedfunctions](https://github.com/nicklockwood/SwiftFormat/blob/main/Rules.md#blanklinesbetweenchainedfunctions) rule removing the blank lines again.

Arguably the blankLinesAroundMark rule should take precedence here, but also it's pretty weird to have a MARK comment in the middle of a chain of functions!

**Comment by nicklockwood** [comment]
@tciuro fixed in 0.54.2
